maint/towncrier_rst_validator.py

```diff
@@ -0,0 +1,103 @@
+from subprocess import STDOUT, check_output
+import argparse
+import os
+
+manual_mode = False
+parser=argparse.ArgumentParser()
+
+parser.add_argument("--pull_request_id", required=True, type=int)
+parser.add_argument("--manual", action="store_true", default=False)
+
+args=parser.parse_args()
+pr_id = args.pull_request_id
+manual_mode = args.manual
+
+rst_dir = "docs/upcoming_changes"
+types_of_changes = ["highlight",
+                    "np_support",
+                    "deprecation",
+                    "expired",
+                    "compatibility",
+                    "cuda",
+                    "new_feature",
+                    "improvement",
+                    "performance",
+                    "change",
+                    "doc",
+                    "infrastructure",
+                    "bug_fix"]
+
+def list_rst_filename() -> str:
+    if manual_mode:
+        all_files = os.listdir(rst_dir)
+        rst_file = [rst_dir + "/" +file for file in all_files if 
+                    file.startswith(str(pr_id) + ".")]
+    else:
+        output = check_output(
+            ["git", "diff", "--name-only", "origin/main"],
+            encoding="utf-8",
+            stderr=STDOUT,
+        )
+        all_files = output.strip().splitlines()
+        rst_file = [file for file in all_files if 
+                    file.startswith(rst_dir + "/" + str(args.pull_request_id) + ".")]
```

Comment:
RE: https://github.com/numba/numba/pull/9335#discussion_r1418113764

> print content of rst_file

I actually meant doing something like `print("found rst_files from git diff:", rst_file)` after this line.

For example, if you run: `python maint/towncrier_rst_validator.py --pull_request_id=9048` on this PR branch, you get:

```
Traceback (most recent call last):
  File "/Users/siu/dev/numba/maint/towncrier_rst_validator.py", line 50, in <module>
    file = list_rst_filename()
  File "/Users/siu/dev/numba/maint/towncrier_rst_validator.py", line 44, in list_rst_filename
    assert len(rst_file) == 1, f"No valid .rst file found in {rst_dir} for the given" + \
AssertionError: No valid .rst file found in docs/upcoming_changes for the given Pull Request ID: 9048. Valid .rst file should  have name starting with the respective Pull Request ID. (see numba/docs/upcoming_changes/README for more details)
```

I was confused that it did't recognize `./docs/upcoming_changes/9048.bug_fix.rst`. The actual problem was that the file is not part of the modified files; not that the filename is invalid.

Reply:
Added the suggested printing convention in [f158a19](https://github.com/numba/numba/pull/9335/commits/f158a197ae4f2f1b056b51ca8ce8b7b586707ed0)